aea/skills/base.py

```diff
@@ -912,3 +765,356 @@ def _print_warning_message_for_non_declared_skill_components(
+        parent_skill_component_types = list(
+            filter(
+                lambda class_: class_ in (Handler, Behaviour, Model),
+                skill_component_type.__mro__,
+            )
+        )
+        enforce(
+            len(parent_skill_component_types) == 1,
+            f"Class {skill_component_type.__name__} in module {skill_component_type.__module__} is not allowed to inherit from more than one skill component type. Found: {parent_skill_component_types}.",
+        )
+        return cast(
+            Type[Union[Handler, Behaviour, Model]], parent_skill_component_types[0]
+        )
+
+    def _match_class_and_configurations(
+        self,
+        component_classes_by_path: Dict[Path, Set[Tuple[str, Type[SkillComponent]]]],
+        declared_component_classes: Dict[
+            _SKILL_COMPONENT_TYPES, Dict[str, SkillComponentConfiguration]
+        ],
+    ) -> List[_SkillComponentLoadingItem]:
+        """
+        Match skill component classes to their configurations.
+
+        Given a class of a skill component, we can disambiguate it in three ways:
+        - by its name
+        - by its type (one of 'Handler', 'Behaviour', 'Model')
+        - whether the user has set the 'file_path' field.
+        If one of the skill component cannot be disambiguated, we raise error.
+
+        In this function, the above criteria are applied in that order.
+
+        :param component_classes_by_path:
+        :return: None
+        """
+        result: List[_SkillComponentLoadingItem] = []
+
+        class_index: Dict[
+            str, Dict[_SKILL_COMPONENT_TYPES, Set[Type[SkillComponent]]]
+        ] = {}
+        used_classes = set()
+        not_resolved_configurations: Dict[
+            Tuple[_SKILL_COMPONENT_TYPES, str], SkillComponentConfiguration
+        ] = {}
+
+        # populate indexes
+        for _path, component_classes in component_classes_by_path.items():
+            for (component_classname, _component_class) in component_classes:
+                type_ = self._get_skill_component_type(_component_class)
+                class_index.setdefault(component_classname, {}).setdefault(
+                    type_, set()
+                ).add(_component_class)
+
+        for component_type, by_id in declared_component_classes.items():
+            for component_id, component_config in by_id.items():
+                path = component_config.file_path
+                class_name = component_config.class_name
+                if path is not None:
+                    classes_in_path = component_classes_by_path[path]
+                    component_class_or_none: Optional[Type[SkillComponent]] = next(
+                        (
+                            actual_class
+                            for actual_class_name, actual_class in classes_in_path
+                            if actual_class_name == class_name
+                        ),
+                        None,
+                    )
```

Comment:
note: cannot have more than one classes with the same name in the same path (the last defined will overwrite the previous ones)